src/main/kotlin/com/github/avrokotlin/avro4k/encoder/PolymorphicEncoder.kt

```diff
@@ -0,0 +1,78 @@
+package com.github.avrokotlin.avro4k.encoder
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.modules.SerializersModule
+
+
+@ExperimentalSerializationApi
+abstract class PolymorphicEncoder(
```

Comment:
Previously was `UnionEncoder`, but since we can now encode a string with schema `["string", "int"]` thanks to the new resolve system, we can also encode polymorphic types but with an instance that only matches the non-union schema (e.g. we can write a polymorphic type `Parent` where children are `Child1` and `Child2`, with the non-union schema `{type: Child1}` only if the given instance is type of Child1. If it's Child2, then it's failing.).

It's totally outside the avro world, and could be used by any other format. This class put clear methods `encodeSerialName` and `encodeValue` instead of respectively unclear `encodeStringElement` and `encodeSerializableElement` so adds lisibility when using polymorphic encoder.